applications/GeoMechanicsApplication/custom_constitutive/truss_backbone_constitutive_law.cpp

```diff
@@ -0,0 +1,228 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//
+
+// Project includes
+#include "truss_backbone_constitutive_law.h"
+#include "geo_mechanics_application_variables.h"
+#include "includes/properties.h"
+
+namespace Kratos
+{
+ConstitutiveLaw::Pointer TrussBackboneConstitutiveLaw::Clone() const
+{
+    return make_shared<TrussBackboneConstitutiveLaw>(*this);
+}
+
+void TrussBackboneConstitutiveLaw::GetLawFeatures(Features& rFeatures)
+{
+    rFeatures.mStrainSize     = 1;
+    rFeatures.mSpaceDimension = 3;
+}
+
+double& TrussBackboneConstitutiveLaw::GetValue(const Variable<double>& rThisVariable, double& rValue)
+{
+    if (rThisVariable == ACCUMULATED_STRAIN) {
+        rValue = mAccumulatedStrain;
+    } else {
+        rValue = BaseType::GetValue(rThisVariable, rValue);
+    }
+
+    return rValue;
+}
+
+void TrussBackboneConstitutiveLaw::SetValue(const Variable<double>& rThisVariable,
+                                            const double&           rValue,
+                                            const ProcessInfo&      rCurrentProcessInfo)
+{
+    if (rThisVariable == ACCUMULATED_STRAIN) {
+        mAccumulatedStrain = rValue;
+    } else {
+        BaseType::SetValue(rThisVariable, rValue, rCurrentProcessInfo);
+    }
+}
+
+double& TrussBackboneConstitutiveLaw::CalculateValue(ConstitutiveLaw::Parameters& rParameterValues,
+                                                     const Variable<double>&      rThisVariable,
+                                                     double&                      rValue)
```

Comment:
No, unfortunately we can't change that, since this member function is an `override`. So the function signature may not be changed. The difficulty with this inherited interface is that the result is returned through the parameter list (as you correctly noticed) as well as by using the return type.